src/zcash/Proof.cpp

```diff
@@ -163,6 +163,10 @@ curve_G2 CompressedG2::to_libsnark_g2() const
 
     assert(r.is_well_formed());
 
+    if (alt_bn128_modulus_r * r != curve_G2::zero()) {
+        throw std::runtime_error("point is not in G2");
+    }
+
```

Comment:
Should ``is_well_formed`` include this check, rather than making it separately?

Reply:
Except that I described it incorrectly in the spec. The check itself is correct. Will fix the spec.